pkg/scheduler/framework/plugins/noderesources/topology_match.go

```diff
@@ -65,22 +65,27 @@ func (tm *TopologyMatch) Name() string {
 }
 
 func filter(containers []v1.Container, nodes []topologyv1alpha1.NUMANodeResource) *framework.Status {
-	for _, container := range(containers) {
+	for _, container := range containers {
 		bitmask := bm.NewEmptyBitMask()
-		for resource, quantity := range(container.Resources.Requests) {
+		var resourceBitmasks []bm.BitMask
+		for resource, quantity := range container.Resources.Requests {
+			// Skipping memory as memory manager not integrated and does not provide topology hints
+			// TODO: Update when Memory Manager is integrated
+			if resource == v1.ResourceMemory || strings.HasPrefix(string(resource), string(v1.ResourceHugePagesPrefix)) {
```

Comment:
I would prefer instead of skipping something  like that.
Set it in case of memory and another corner cases. Because in this case we can use this numa node...
you may omit QoS in you change, I'll add it later.
                        for _, numaNode := range(nodes) {
                                numaQuantity, ok := numaNode.Resources[resource]
-                               if !ok || numaQuantity.Cmp(quantity) < 0 {
+                               // we don't expect numaQuantity 0 here
+                               if !ok {
                                        continue
                                }
-                               resourceBitmask.Add(numaNode.NUMAID)
+
+                               // condition is following:
+                               // 1. set numa node as possible node if resource is cpu and it's not guaranteed QoS, since cpu will flow
+                               // 2. set numa node as possible node if resource is memory (until memory manager will not be merged and
+                               // memory will not be provided in CRD
+                               // 3. otherwise check amount of resources
+                               if resource == v1.ResourceCPU && qos != v1.PodQOSGuaranteed  ||
+                                       resource == v1.ResourceMemory ||
+                                       numaQuantity.Cmp(quantity) >= 0 {
+                                       // cpu will not be pinned, we don't need exclusive cpu
+                                       resourceBitmask.Add(numaNode.NUMAID)
+                               }

Reply:
Can you please update your comment by adding code you have above to a code block if you don't mind, it will help in the readability of your comment.

Reply:
Also commit message like: "support corner case with memory", I think is better.

Reply:
and it's 2 topic in this commit: code style fixes (I guess it's from hack/verify-fofmt/lint), memory case.

Reply:
I ran gofmt and goimports for fomatting

Reply:
I tested these changes and these are leading for pods ending up in a pending state, working on a working implementation, will push that soon and remove the WIP tag

Reply:
The problem in that the statement in case of memory/hugepages
```
numaQuantity, ok := numaNode.Resources[resource]
  
if !ok || numaQuantity.Cmp(quantity) < 0 {
continue
}
```
skips to the next iteration as numaNode.Resources["memory"]  evaluates to ok being false. Checking for memory is useless later because control neve reaches that part in case of memory

A way of solving this is setting all the bits of resourceMask in case memory and hugepages prior at the time of initialization. I have updated the code based on that and have tested it. Let me know if that looks okay to you.